lms/djangoapps/canvas_integration/client.py

```diff
@@ -0,0 +1,140 @@
+import time
+from urllib.parse import urljoin
+import requests
+
+from django.conf import settings
+
+from canvas_integration.constants import DEFAULT_ASSIGNMENT_POINTS
+
+
+class CanvasClient:
+    def __init__(self, canvas_course_id):
+        self.session = self.get_canvas_session()
+        self.canvas_course_id = canvas_course_id
+
+    @staticmethod
+    def get_canvas_session():
+        """
+        Create a request session with the access token
+        """
+        session = requests.Session()
+        session.headers.update({
+            "Authorization": "Bearer {token}".format(token=settings.CANVAS_ACCESS_TOKEN)
+        })
+        return session
+
+    def list_canvas_enrollments(self):
+        """
+        Fetch canvas enrollments. This may take a while, so don't run in the request thread.
+
+        Args:
+            canvas_course_id (int): The canvas id for the course
+
+        Returns:
+            dict: Email addresses mapped to canvas user ids for all enrolled users
+        """
+        email_id_map = {}
+        url = urljoin(
+            settings.CANVAS_BASE_URL,
+            "/api/v1/courses/{course_id}/enrollments".format(course_id=self.canvas_course_id)
+        )
+        while url:
```

Comment:
Can we refactor this custom pagination code so we can use this on `list_canvas_assignments` and `list_canvas_grades` too?